## Test of other models
As mentioned in the Report, we are loading and testing the four GEMs using Memote. The results are summarized in Table 1 at the bottom of the page.


In [1]:
from cobra.io import read_sbml_model, write_sbml_model
from cobra import Reaction, Metabolite

In [9]:
# Loading of the iMM904 model
iMM904 =  read_sbml_model('data/iMM904.xml.gz')

iMM904

<Model iMM904 at 0x7f1a630fce10>

In [2]:
# Loading of the iND750 model
iND750 =  read_sbml_model('data/iND750.xml.gz')

iND750

<Model iND750 at 0x7f1a65b09080>

In [7]:
# Loading of the Yeast8 model
Yeast8 =  read_sbml_model('data/yeastGEM.xml')

Yeast8

<Model M_yeastGEM_v8__46__4__46__2 at 0x7efc7f88d668>

In [8]:
# Loading of the ecYeast8 model
ecYeast8 =  read_sbml_model('data/ecYeastGEM_prot.xml')

ecYeast8

<Model M_ecYeastGEM_prot_v8__46__3__46__4 at 0x7efc7dbac4a8>

We can now run the memote analysis on all of our models to learn more about their qualities

In [17]:
# iMM904 analysis
#%%time

!memote report snapshot data/iMM904.xml.gz --filename memote/iMM904.html

============================= test session starts ==============================
platform linux -- Python 3.6.12, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /usr/local/lib/python3.6/dist-packages/memote/suite/tests
collected 146 items / 1 skipped / 145 selected                                 

../../../../usr/local/lib/python3.6/dist-packages/memote/suite/tests/test_annotation.py . [  0%]
..FFFFFFFF..FFFFF.FFFFFFFFFFFFFFF..FF..F...F....FFF.FFFFF.FFFF..         [ 44%]
../../../../usr/local/lib/python3.6/dist-packages/memote/suite/tests/test_basic.py . [ 45%]
.....FF.......F...F.FF                                                   [ 60%]
../../../../usr/local/lib/python3.6/dist-packages/memote/suite/tests/test_biomass.py . [ 60%]
F....FF.F                                                                [ 67%]
../../../../usr/local/lib/python3.6/dist-packages/memote/suite/tests/test_consistency.py . [ 67%]
..ssssssssssssssssFFFFFF.FFF                                             [ 86%]

In [16]:
# iND750 analysis
#%%time

!memote report snapshot data/iND750.xml.gz --filename memote/iND750.html

============================= test session starts ==============================
platform linux -- Python 3.6.12, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /usr/local/lib/python3.6/dist-packages/memote/suite/tests
collected 146 items / 1 skipped / 145 selected                                 

../../../../usr/local/lib/python3.6/dist-packages/memote/suite/tests/test_annotation.py . [  0%]
..FFFFFFFF..FFFF..FFFFFFFFFFFFFFF..FF..F........FFF.FFFFF.FFFF..         [ 44%]
../../../../usr/local/lib/python3.6/dist-packages/memote/suite/tests/test_basic.py . [ 45%]
.....FF.......F...F.FF                                                   [ 60%]
../../../../usr/local/lib/python3.6/dist-packages/memote/suite/tests/test_biomass.py . [ 60%]
F....FF.F                                                                [ 67%]
../../../../usr/local/lib/python3.6/dist-packages/memote/suite/tests/test_consistency.py . [ 67%]
..ssssssssssssssss.FFFFF.FFF                                             [ 86%]

In [ ]:
# Yeast8 analysis
#%%time

!memote report snapshot data/yeastGEM.xml --filename memote/yeastGEM.html

============================= test session starts ==============================
platform linux -- Python 3.6.12, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /usr/local/lib/python3.6/dist-packages/memote/suite/tests
collected 146 items / 1 skipped / 145 selected                                 

../../../../usr/local/lib/python3.6/dist-packages/memote/suite/tests/test_annotation.py . [  0%]
..FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF.FFF.FF..FF.F..F.FFFFFFFFFFFFF.         [ 44%]
../../../../usr/local/lib/python3.6/dist-packages/memote/suite/tests/test_basic.py . [ 45%]
.....F........F.F.F.FF                                                   [ 60%]
../../../../usr/local/lib/python3.6/dist-packages/memote/suite/tests/test_biomass.py . [ 60%]
F.....F.F                                                                [ 67%]
../../../../usr/local/lib/python3.6/dist-packages/memote/suite/tests/test_consistency.py F [ 67%]
FFssssssssssssssssFFFFFF.FFF                                             [ 86%]

In [ ]:
# ecYeast8 analysis
#%%time

!memote report snapshot data/ecYeastGEM_prot.xml --filename memote/ecYeastGEM_prot.html

We can now make ourself an overview of our models:

**Table 1: Overview of metabolic models**

| Model | # Metabolites | # Reactions | # Compartments | Memote analysis |
| --- | --- | --- | --- | --- |
| iMM904 | 1226 | 1577 | 8 | 85% |
| iND750 | 1059 | 1266 | 8 | 86% |
| Yeast8 | 2742 | 4058 | 14 | 65% |
| ecYeast8 | 4180 | 8144 | 14 | 16% |